### PR TITLE
Create painters_partition.cpp

### PR DESCRIPTION
**Pull request by vaishnavijha**
#30  solved by using binary search algorithm on painters partition problem

### CODE REVIEW

**Comment by Anshuvats** [comment]
Please mention at least one sample test case in the description of the code.

File: Contributor Corner/vaishnavi/painters_partition.cpp [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+Calculate and return minimum time required to paint all boards under the constraints that any painter will only paint contiguous sections of board.
+
+2 painters cannot share a board to paint. That is to say, a board
+cannot be painted partially by one painter, and partially by another.
+A painter will only paint contiguous boards. Which means a
+configuration where painter 1 paints board 1 and 3 but not 2 is
+invalid.
+*/
+
```
Comment (Abhijit2505) [inline Contributor Corner/vaishnavi/painters_partition.cpp:17]: Please add a sample test case which can be shown to users.

File: Contributor Corner/vaishnavi/painters_partition.cpp [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+//solution by binary search algorithm
+#include<bits/stdc++.h>
+using namespace std;
+bool isPossible(int mid, vector<int> &c, int a) {
```
Comment (Abhijit2505) [inline Contributor Corner/vaishnavi/painters_partition.cpp:21]: Do comment on what the function `isPossible()` does in short.

File: Contributor Corner/vaishnavi/painters_partition.cpp [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        if(Time > mid) {
+            Time = c[i];
+            numPainters++;
+            if(numPainters > a) return false;
+        }
+    }
+    return true;
+}
+
+int paint(int A, int B, vector<int> &C) {
```
Comment (Abhijit2505) [inline Contributor Corner/vaishnavi/painters_partition.cpp:34]: Do comment on what `paint()` do in short.

**Review by Abhijit2505 — CHANGES_REQUESTED** [review]

**Comment by Abhijit2505** [comment]
> Please mention at least one sample test case in the description of the code.

Well thanks for reviewing it unofficially, but we encourage you to add some lines of code instead, reviewing will be done by the mentor and the team.

File: Contributor Corner/vaishnavi/painters_partition.cpp (hunk 1 above)
Comment (vaishnavijha) [reply Contributor Corner/vaishnavi/painters_partition.cpp:38]: I have added 2 test cases  in the comment of the question .Please review it.

File: Contributor Corner/vaishnavi/painters_partition.cpp (hunk 2 above)
Comment (vaishnavijha) [reply Contributor Corner/vaishnavi/painters_partition.cpp:48]: I have commented  now about the work of every function in the code.

**Comment by vaishnavijha** [comment]
Please review the requested changes made in the code .

**Comment by Abhijit2505** [comment]
@all-contributors add @vaishnavijha for code

**Pull request merged**